### PR TITLE
Enable msvc for run-make/rust-lld

### PR DESCRIPTION
**Pull request by ChrisDenton**
This is simply a matter of using the right argument for lld-link.

As a bonus, I also fixed a typo.

try-job: i686-msvc
try-job: x86_64-msvc

### CODE REVIEW

**Comment by lqd** [comment]
I don’t remember we’ve implemented self-contained linking other than by using a c/c++ compiler driver link-arg, while msvc needs to change the linker, so it’ll be interesting to see the results.

**Comment by lqd** [comment]
r=me if it works tho

**Comment by ChrisDenton** [comment]
Success!

@bors r=lqd rollup

**Comment by lqd** [comment]
Ah I think I know why it works even though we didn’t purposefully implement self-contained linking on non-unixy platforms: we can change the linker according to the linker flavor, and we currently use an lld linker-flavor as an implementation detail of self-contained linking. I guess this test will prevent us from being lazy when we refactor this away 😅 .

**Comment by ChrisDenton** [comment]
Failed in rollup... on Apple??? Huh.

https://github.com/rust-lang/rust/pull/128668#issuecomment-2268150106

File: tests/run-make/rust-lld/rmake.rs [hunk 1]
```diff
@@ -2,23 +2,25 @@
...
 
 use std::process::Output;
 
 use run_make_support::regex::Regex;
-use run_make_support::rustc;
+use run_make_support::{is_msvc, rustc};
 
 fn main() {
+    // lld-link is used if msvc, otherwise a gnu-compatible lld is used.
+    let linker_version_flag = if is_msvc() { "--version" } else { "Wl,-v" };
```
Comment (ChrisDenton) [inline tests/run-make/rust-lld/rmake.rs:14]: Oh, I made a type of my own. I forgot the dash `-`. Silly me.

**Comment by ChrisDenton** [comment]
Let's try that again...

@bors r=lqd

**Pull request merged**